plugins/meta/iptables-allow/iptables_allow.go

```diff
@@ -0,0 +1,210 @@
+// Copyright 2016 CNI authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This is a "meta-plugin". It reads in its own netconf, it does not create
+// any network interface but just changes the network sysctl.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/coreos/go-iptables/iptables"
+
+	"github.com/containernetworking/cni/pkg/skel"
+	"github.com/containernetworking/cni/pkg/types"
+	"github.com/containernetworking/cni/pkg/types/current"
+	"github.com/containernetworking/cni/pkg/version"
+)
+
+// IptAllowConf represents the iptables allow configuration.
+type IptAllowConf struct {
+	types.NetConf
+
+	// IfName is the interface name to apply rules for; if not given the
+	// interface name will be pulled from the first host interface of
+	// the PrevResult (eg, the first one without a sandbox key).
+	IfName string `json:"ifName,omitempty"`
+	// AdminChainName is an optional name to use instead of the default
+	// admin rules override chain name that includes the interface name.
+	AdminChainName string `json:"adminChainName,omitempty"`
+
+	RawPrevResult map[string]interface{} `json:"prevResult,omitempty"`
+	PrevResult    *current.Result        `json:"-"`
+}
+
+func parseConf(data []byte) (*IptAllowConf, error) {
+	conf := IptAllowConf{}
+	if err := json.Unmarshal(data, &conf); err != nil {
+		return nil, fmt.Errorf("failed to load netconf: %v", err)
+	}
+
+	// Parse previous result.
+	if conf.RawPrevResult != nil {
+		resultBytes, err := json.Marshal(conf.RawPrevResult)
+		if err != nil {
+			return nil, fmt.Errorf("could not serialize prevResult: %v", err)
+		}
+		res, err := version.NewResult(conf.CNIVersion, resultBytes)
+		if err != nil {
+			return nil, fmt.Errorf("could not parse prevResult: %v", err)
+		}
+		conf.RawPrevResult = nil
+		conf.PrevResult, err = current.NewResultFromResult(res)
+		if err != nil {
+			return nil, fmt.Errorf("could not convert result to current version: %v", err)
+		}
+	}
+
+	return &conf, nil
+}
+
+func getPrivChainRules(intf string) [][]string {
+	var rules [][]string
+	rules = append(rules, []string{"-i", intf, "!", "-o", intf, "-j", "ACCEPT"})
+	rules = append(rules, []string{"-i", intf, "-o", intf, "-j", "ACCEPT"})
```

Comment:
Why do we need two rules here?  Can't we just not match on -o at all?

Reply:
@matthewdupre consolidated the two rules, you're right we don't need the double-match.  The docker code that was adding both was doing it from two unrelated pieces of code which probably aren't aware of each other.